### ISSUE TITLE
This repo is dead! Or is it?

### ISSUE THREAD

**Issue by maxfrei750** [matterport/Mask_RCNN#936]
![Dead parrot sketch](http://3.bp.blogspot.com/-QsJRwCx_vvQ/UWQcu_uPb8I/AAAAAAAATJs/THt3Xmfz17A/s640/Dead+parrot+2.jpg "Dead parrot sketch")

Now that's indeed the question... is it just resting or is it actually dead? The last commit is over 2 months old and @waleedka's activity has been scarce for quite some time in general. I hope that the reason for this are perhaps a long holiday, a bunch of more interesting things in his life than this repo, that he won the lottery or another merry reason and not that something bad happened.

This repo has been maintained by @waleedka  for quite a long time and he never asked for anything in return. Therefore, I'm very grateful. However, for the very same reason, I think that it would be a lot of wasted potential, if this repo was no longer maintained by anybody. That's why I'd like to discuss the situation - because perhaps some of you might know more than me - and also possible solutions, if this repo was actually dead. After all, there are already quite a number of pull requests which should IMHO be accepted or closed (e.g. #390, #926, #925, #878, #877, #873, #823, #798, #740). 

Off the top pf my head I can think of two solutions (given that the repo was actually no longer maintained):
1. @waleedka  assigns somebody as a contributor, so that he/she can accept/close pull requests, etc.. I think somebody who already made some important contributions would be a good choice (e.g. @Borda or @philferriere). Of course that person would also have to have the time to maintain the project.
2. Somebody could mirror this project and maintain the project from there on.

Personally, I would strongly prefer the first solution or even better, that @waleedka is just on holiday and returns to keep up the great work.

**Comment by Borda** [comment]
Regarding so, there is still open PR #390 which would significantly simplify community contributions... :)

**Comment by maxfrei750** [comment]
Hurray! @waleedka has returned! Long live the repo! 😄 However, it would still be a good idea to discuss how we could improve the development of this repo by adding automated testing and have one ore more additional maintainers.

**Comment by luxedo** [comment]
Is it?!

**Comment by Borda** [comment]
isn't it? :)

**Comment by maxfrei750** [comment]
That's a really hard question. But I don't think so. To be honest, I thought about the possibilities of a fork more than once. There are so many things that might be improved. However, I don't think, that I am at all suited for that job. @Borda How about you? I'd be glad to help out, wherever I can.

**Comment by Borda** [comment]
I already forked it and made some maintenance... That tried to \ping @waleedka to merge these changes, but it is silence for a while... the kind of disadvantage of the fork (so far I know) is the child do not have access to the listed issues nor proposed PRs... Me personally I would welcome a decision if this repo is alive or not and if not I may take some responsibility and continue maintenance in my own fork... What do you think?

**Comment by maxfrei750** [comment]
You're totaly right that the best solution would be that matterport/@waleedka opened the repo to maintainers and collaborators. It is also in their own interest, as it generates attention for their other projects. However, I already proposed the opening of the repo a while ago and it was ignored, so apparently there is no interest in doing so.

Concerning the PRs: You are totaly right, however, it is easy to add the branches associated with the PRs to your fork and to merge them. And at some point of time in the future, when your fork is advancing further, people might start to make their PRs at your repo.

Concerning the issues: From my impression, many issues are rather questions about the usage than hints to actual bugs or feature requests. If there is an issue that is really interesting to you or another maintainer, it might be a good idea to contact the original issuer and inform them that you would like to solve their problem in your fork. I think it might be very likely that people will collaborate, because they want their problem to be solved. And again, after a while people might start to add their issues at your fork. "Loosing" all the issues might even be a good thing. After all there are nearly 1000 issues. Great chance to get rid of all the clutter. 😅

BTW, I think it is great that you would be willing to further improve the project and I would love to contribute.

**Comment by Borda** [comment]
@maxfrei750 how did you contact them, because of I not aware they were paying to any mentions or comments related to this repo for many months... so try to reach them via regular e-mail?

According to the GitHub profile, there are no members of the company listed - https://github.com/orgs/matterport/people

**Comment by maxfrei750** [comment]
@Borda No, I just proposed it in this issue, directly after @waleedka returned for a few commits and by doing so temporarily ended the discussion. An email sounds like a good idea, so I just wrote them one. Let's see if and how they react. They cannot really be interested in loosing this repo. I mean look at it, it's hugely successful. I'll keep you updated.

**Comment by Borda** [comment]
Well, maybe it make sense to have a look at torchvision... https://github.com/pytorch/vision

**Comment by Borda** [comment]
@maxfrei750 did you get any feedback? an alternative would be making a successor from our fork... What do you think? @waleedka @luxedo ^^
Just had a chat with Matterport support nad they will pas the question to their development team... :)

**Comment by Borda** [comment]
@waleedka @keineahnung2345 @philferriere @Cpruce @imgyuri @ps48 @DingkunLiu ^^

**Comment by maxfrei750** [comment]
@Borda I didn't get any response so far. Let's see what the dev team says...

**Comment by Borda** [comment]
it seems that there is a FB group... https://www.facebook.com/groups/MatterportOfficial
open post - https://www.facebook.com/groups/MatterportOfficial/permalink/2447306635376328/

**Comment by akrsrivastava** [comment]
So, whats the latest on this. Seems like this is a dead repo now.

**Comment by maxfrei750** [comment]
Yeah, pretty much. I'd recommend: 
https://github.com/pytorch/vision
or
https://github.com/facebookresearch/detectron2/

**Comment by ppwwyyxx** [comment]
Or if you're looking for a tensorflow version, https://github.com/tensorpack/tensorpack/tree/master/examples/FasterRCNN has also been there for 2+ years.

**Comment by maxfrei750** [comment]
@ppwwyyxx Thanks for the tip.

I think it is time to bury this issue. 
**The repo is dead! Long live the repos!**

**Comment by bertsky** [comment]
Thanks @maxfrei750 and everyone who answered for casting some light on this sad circumstance!

Still, IMHO this issue should be kept open until @waleedka either archives the repo (so others do not get trapped), transfers its ownership, or accords some contributors with collaborator priviledges.

**Comment by maxfrei750** [comment]
@bertsky Agreed.
